test/e2e/suite/issuers/acme/certificate/http01.go

```diff
@@ -270,14 +269,8 @@ var _ = framework.CertManagerDescribe("ACME Certificate (HTTP01)", func() {
 		cert, err = certClient.Update(context.TODO(), cert, metav1.UpdateOptions{})
 		Expect(err).NotTo(HaveOccurred())
 
-		By("Waiting for the Certificate to be not ready")
-		_, err = h.WaitForCertificateNotReady(f.Namespace.Name, certificateName, time.Minute*5)
-		Expect(err).NotTo(HaveOccurred())
-
-		By("Waiting for the Certificate to become ready & valid")
-
-		By("Waiting for the Certificate to be issued...")
-		err = f.Helper().WaitCertificateIssued(f.Namespace.Name, certificateName, time.Minute*5)
```

Comment:
Why is this being removed?

Reply:
Because the non-ready state will self-resolve. The non-ready state could be not observed/ skipped. Instead, using the ObservedGeneration, we can wait for a Ready state even if the current state is already Ready.

Reply:
Understood, makes sense. Can you add a comment to the effect that we are waiting for a ready condition where the observed generation is the same generation of the latest spec that we just updated?